src/unread/UnreadCards.js

```diff
@@ -42,7 +42,9 @@ class UnreadCards extends PureComponent<Props> {
   };
 
   handleTopicPress = (stream: string, topic: string) => {
-    this.props.dispatch(doNarrow(topicNarrow(stream, topic)));
+    setTimeout(() => {
```

Comment:
I thinks it will be better if we move this logic to `Touchable` component itself, as it will be then been applied to all the press throughout the app.

Thoughts? @borisyankov @gnprice @zulip/react-native-committers

Reply:
I think we should even move it to `doNarrow` as grepping `dispatch(doNarrow` yields 24 results and many (all?) of them will benefit from giving a chance to the UI to respond.

I would extract everything from `dispatch(switchNarrow(narrow));` to the end of the file in a new function, maybe called something like `switchNarrowAndFetch()` and then call this function from `setTimeout`.

Reply:
What I just suggested is quite enough, but as an additional reading, this is a great read (by the author of Redux):
https://stackoverflow.com/questions/35411423/how-to-dispatch-a-redux-action-with-a-timeout